kernel/rtlil.h

```diff
@@ -152,6 +152,8 @@ namespace RTLIL
 
 			log_assert(p[0] == '$' || p[0] == '\\');
 			log_assert(p[1] != 0);
+			for (const char *c = p; *c; c++)
+				log_assert(*c > ' ');
```

Comment:
Two things from JF:

- what about characters `>= 0x7f`? Should the check just be `*c > ' ' && *c < 0x7f`?
- this (and probably the existing two asserts) would be better positioned in the insert path (after the following `if`), so that we don't run it for every lookup

Reply:
> * what about characters `>= 0x7f`? Should the check just be `*c > ' ' && *c < 0x7f`?

I see no particular reason to forbid extended charset identifiers at the level of Yosys core. Yosys doesn't care, some backends probably care but backends have to make sure they properly escape various special characters in identifiers anyway.

That said, I wouldn't oppose prohibiting them if there's consensus that it's the right thing.

> this (and probably the existing two asserts) would be better positioned in the insert path (after the following if), so that we don't run it for every lookup

Will do.

Reply:
There's a little problem here though: since `char` may or may not be signed, you may or may not be already rejecting extended charset identifiers (because they could be negative) — if we want to allow them, the condition needs changing to `*c > ' ' || *c < 0`

Reply:
Argh, I forgot about that. Yeah, I actually meant `(unsigned)*c > (unsigned)' '`.

Reply:
Done